sdk/identity/identity/src/util/validateMultiTenant.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions } from "@azure/core-auth";
+
+/**
+ * @internal
+ */
+export const multiTenantError = new Error(
+  "Multi-tenant authentication was attempted, but multi-tenant authentication was not enabled in this credential instance."
+);
+
+/**
+ * Verifies whether locally assigned tenants are equal to tenants received through getToken.
+ * Returns the appropriate tenant.
+ * @internal
+ */
+export function processMultiTenantRequest(
+  tenantId?: string,
+  allowMultiTenantAuthentication?: boolean,
+  getTokenOptions?: GetTokenOptions
+): string | undefined {
+  if (
+    !allowMultiTenantAuthentication &&
+    getTokenOptions?.tenantId &&
+    tenantId &&
+    getTokenOptions.tenantId !== tenantId
+  ) {
+    throw multiTenantError;
+  }
+  if (allowMultiTenantAuthentication && getTokenOptions?.tenantId) {
```

Comment:
wouldn't these always be true, since otherwise the if statement above would be get hit and we'd throw?

Reply:
No no, the statement above only throws if `allowMultiTenantAuthentication` is false. This is the case when it is true and the getTokenOptions tenant is defined, in which case we take preference on that tenant instead of the tenant configured in the credential.

Reply:
yep yep